### PR TITLE
Added preview flag to black

### PR DESCRIPTION
**Pull request by KareemMAX**
This is a test of the `--preview` flag of `black`, to see if you can add it to our linting process or not

### CODE REVIEW

File: ivy/functional/backends/jax/experimental/layers.py [hunk 1]
```diff
@@ -597,9 +597,7 @@ def interpolate(
     mode = (
         "nearest"
         if mode == "nearest-exact"
-        else "bicubic" "bicubic"
-        if mode == "bicubic_tensorflow"
-        else mode
+        else "bicubicbicubic" if mode == "bicubic_tensorflow" else mode
```
Comment (KareemMAX) [inline ivy/functional/backends/jax/experimental/layers.py:600]: This is one of the simplification of string preprocessing of black, I guess it was a bug in our initial code.

File: ivy/data_classes/container/base.py [hunk 2]
```diff
@@ -1120,8 +1120,10 @@ def cont_from_disk_as_hdf5(
         """
         ivy.utils.assertions.check_exists(
             h5py,
-            message="You must install python package h5py in order to load hdf5 \
-            files from disk into a container.",
+            message=(
+                "You must install python package h5py in order to load hdf5            "
+                " files from disk into a container."
+            ),
```
Comment (KareemMAX) [inline ivy/data_classes/container/base.py:1126]: This is on of the artifacts of string preprocessing as well, I'm not sure of the behavior of `\ ` and I couldn't find anything online. Maybe it's a bug in our code as well

File: ivy/data_classes/container/statistical.py [hunk 3]
```diff
@@ -79,9 +79,11 @@ def min(
...
         return self.cont_handle_inplace(
             self.cont_map(
-                lambda x_, _: ivy.min(x_, axis=axis, keepdims=keepdims)
-                if ivy.is_array(x_)
-                else x_,
+                lambda x_, _: (
+                    ivy.min(x_, axis=axis, keepdims=keepdims)
+                    if ivy.is_array(x_)
+                    else x_
+                ),
```
Comment (KareemMAX) [inline ivy/data_classes/container/statistical.py:86]: This is improved parentheses managment, black wraps `x if c else y` with parentheses, and it looks like we have a lot of those in our code. I guess it's some improved readablity (at least for me).

File: ivy/functional/backends/torch/experimental/elementwise.py [hunk 4]
```diff
@@ -84,9 +84,9 @@ def trapz(
     else:
         if dx is not None:
             TypeError(
-                "trapezoid() received an invalid combination of arguments - got\
-            (Tensor, Tensor, int), but expected one of: *\
-            (Tensor y, Tensor x, *, int dim) * (Tensor y, *, Number dx, int dim)"
+                "trapezoid() received an invalid combination of arguments - got        "
+                "    (Tensor, Tensor, int), but expected one of: *            (Tensor"
+                " y, Tensor x, *, int dim) * (Tensor y, *, Number dx, int dim)"
```
Comment (KareemMAX) [inline ivy/functional/backends/torch/experimental/elementwise.py:89]: This is one other example of the usage of `\ `

File: ivy/data_classes/container/base.py (hunk 2 above)
Comment (CatB1t) [reply ivy/data_classes/container/base.py:1126]: I think the correct one should be 
```suggestion
            message=(
                "You must install python package h5py in order to load hdf5" \
                " files from disk into a container."
            ),
```

File: ivy/functional/backends/paddle/random.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
 
 @with_unsupported_dtypes(
-    {"2.4.2 and below": ("int8")},
+    {"2.4.2 and below": "int8"},
```
Comment (CatB1t) [inline ivy/functional/backends/paddle/random.py:24]: This should be a tuple, `("int8",)`

File: ivy/functional/backends/tensorflow/experimental/creation.py [hunk 6]
```diff
@@ -40,7 +40,7 @@ def triu_indices(
     return tuple(tf.convert_to_tensor(ret, dtype=tf.int64))
 
 
-@with_supported_dtypes({"1.11.0 and below": ("int32")}, backend_version)
+@with_supported_dtypes({"1.11.0 and below": "int32"}, backend_version)
```
Comment (CatB1t) [inline ivy/functional/backends/tensorflow/experimental/creation.py:43]: This should be a tuple, ("int32",)

File: ivy/functional/frontends/jax/numpy/statistical.py [hunk 7]
```diff
@@ -401,7 +400,7 @@ def nancumprod(a, axis=None, dtype=None, out=None):
 
 
 @handle_jax_dtype
-@with_unsupported_dtypes({"1.11.0 and below": ("bfloat16")}, "jax")
+@with_unsupported_dtypes({"1.11.0 and below": "bfloat16"}, "jax")
```
Comment (CatB1t) [inline ivy/functional/frontends/jax/numpy/statistical.py:403]: This should be a tuple, `("bfloat16",)`

File: ivy/functional/frontends/tensorflow/math.py [hunk 8]
```diff
@@ -514,7 +514,7 @@ def nextafter(x1, x2, name=None):
 @with_unsupported_dtypes(
     {
         "1.2.0": ("float16", "complex64", "complex128"),
-        "1.8.0 and below": ("float16"),
+        "1.8.0 and below": "float16",
```
Comment (CatB1t) [inline ivy/functional/frontends/tensorflow/math.py:517]: This should be a tuple, `("float16",)`

File: ivy/utils/backend/sub_backend_handler.py [hunk 9]
```diff
@@ -57,21 +57,21 @@ def set_sub_backend(sub_backend_str: str):
 
     if ivy.current_backend_str() not in _backend_to_sub_backends_dict.keys():
         logging.warn(
-            f"backend {ivy.current_backend_str()}\
-                  does not have any supported sub_backends"
+            f"backend {ivy.current_backend_str()}                  does not have any"
```
Comment (CatB1t) [inline ivy/utils/backend/sub_backend_handler.py:60]: I think ideally should be:
```suggestion
            f"backend {ivy.current_backend_str()} does not have any" \
            " supported sub_backends"
```

File: ivy/functional/backends/jax/data_type.py [hunk 10]
```diff
@@ -136,7 +136,7 @@ def broadcast_arrays(*arrays: JaxArray) -> List[JaxArray]:
 
 
 @with_unsupported_dtypes(
-    {"0.3.14 and below": ("complex")},
+    {"0.3.14 and below": "complex"},
```
Comment (CatB1t) [inline ivy/functional/backends/jax/data_type.py:139]: Should be a tuple `("complex",)`

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @KareemMAX, just added a few comments regarding `un/supported_dtypes` decorator, the second argument should be a tuple, otherwise it wouldn't pick it correctly.

**Pull request merged**